Core/Data.EF/Extensions/EntityExtensions.cs

```diff
@@ -31,10 +32,27 @@ public static IQueryable<TEntity> GetByQueryModel<TEntity>(this IQueryable<TEnti
        where TEntity : class, IEntity
     {
         // apply default filters
-        if (typeof(ITagsEntity).IsAssignableFrom(typeof(TEntity)) && !string.IsNullOrEmpty(queryModel.Tag))
+        if (typeof(ITagsEntity).IsAssignableFrom(typeof(TEntity)))
```

Comment:
Instead of chaining `if` statements here, I'd rather use a factory pattern, to reduce complexity and responsibilities of this function:

Pseudocode:
```
var filters = QueryModelFiltersFactory.Create<TEntity>();
foreach (var filter in filters) {
    entities = filter.Filter(entities);
}
```

Something along these lines.

Reply:
This sounds really interesting, but i don't have the knowledge to do it in C#! If you have some time before we'll close this PR to help me, I would really be interested! Otherwise I will consider this comment as solved!

Reply:
Definitely nice, maybe even an improvement, but for now I'd say leave it as is and don't spent time on it.